rust/main/chains/hyperlane-sealevel/src/mailbox.rs

```diff
@@ -508,133 +463,40 @@ impl Mailbox for SealevelMailbox {
         metadata: &[u8],
         _tx_gas_limit: Option<U256>,
```

Comment:
it's a bit confusing to have this function called with `Some(0)` value (from the `process_estimate_costs` result) to then ignore it. I think we'd respect the trait interface better if we returned `gas_limit: None` in `process_estimate_costs`, called `unwrap_or_default` on it when doing the igp check, and feeding this argument into `build_estimated_tx_for_instruction`

Reply:
agree it's weird, though think we should maybe just leave this to a later PR